tests/matrix.rs

```diff
@@ -97,6 +99,26 @@ pub mod matrix2 {
     }
 
     #[test]
+    fn test_sum_matrix() {
+        // This doesn't work because there isn't an impl for &Matrix2<f64>.
+        // let res: Matrix2<f64> = [A, B, C].iter().sum();
+
+        // This type resolves at least, but is ugly.
+        let res: Matrix2<f64> = [A, B, C].iter().cloned().sum();
```

Comment:
Could always do `vec![A, B, C].into_iter().sum()`... not sure if that is any better though... ugh, another allocation. It's only a test though, I guess. No need for the comment if you want to leave it though - I think it is clear enough.